bc_obps/service/tests/test_facility_service.py

```diff
@@ -42,3 +52,101 @@ def test_list_facilities_for_industry_user():
         )
+
+
+class TestGetIfAuthorized:
+    @staticmethod
+    def test_get_if_authorized_cas_user_success():
+        user = baker.make(User, app_role=AppRole.objects.get(role_name="cas_analyst"))
+
+        facility = baker.make(Facility, latitude_of_largest_emissions=5, longitude_of_largest_emissions=5)
+        baker.make(FacilityOwnershipTimeline, operation=operation_baker(), facility=facility)
+
+        result = FacilityService.get_if_authorized(user.user_guid, facility.id)
+        assert result == facility
+
+    @staticmethod
+    def test_get_if_authorized_industry_user_success():
+        user = baker.make(User, app_role=AppRole.objects.get(role_name="industry_user"))
+        operator = operator_baker()
+        baker.make(
+            UserOperator,
+            user_id=user.user_guid,
+            status=UserOperator.Statuses.APPROVED,
+            operator=operator,
+            role=UserOperator.Roles.ADMIN,
+        )
+        owning_operation: Operation = operation_baker(operator.id)
+        facility = baker.make(Facility, latitude_of_largest_emissions=5, longitude_of_largest_emissions=5)
+        baker.make(FacilityOwnershipTimeline, operation=owning_operation, facility=facility)
+
+        result = FacilityService.get_if_authorized(user.user_guid, facility.id)
+        assert result == facility
+
+    @staticmethod
+    def test_get_if_authorized_industry_user_fail():
+        user = baker.make(User, app_role=AppRole.objects.get(role_name="industry_user"))
+        owning_operator = operator_baker()
+        random_operator = operator_baker()
+        baker.make(
+            UserOperator,
+            user_id=user.user_guid,
+            status=UserOperator.Statuses.APPROVED,
+            operator=random_operator,
+            role=UserOperator.Roles.ADMIN,
+        )
+        owning_operation: Operation = operation_baker(owning_operator.id)
+        facility = baker.make(Facility, latitude_of_largest_emissions=5, longitude_of_largest_emissions=5)
+        baker.make(FacilityOwnershipTimeline, operation=owning_operation, facility=facility)
+
+        with pytest.raises(Exception, match=UNAUTHORIZED_MESSAGE):
+            FacilityService.get_if_authorized(user.user_guid, facility.id)
+
+
+class TestCreateFacilityWithOwnership:
+    @staticmethod
+    def test_create_sfo_facility_with_ownership_without_address():
+        user = baker.make(User, app_role=AppRole.objects.get(role_name="industry_user"))
+        payload = {
+            'facility_data': {
+                'name': 'zip',
+                'type': 'Single Facility',
+                'latitude_of_largest_emissions': 5,
+                'longitude_of_largest_emissions': 5,
+            },
+            'operation_id': operation_baker().id,
+        }
+
+        FacilityService.create_facility_with_ownership(user.user_guid, payload)
+        assert len(Facility.objects.all()) == 1
```

Comment:
nit: we could use `Facility.objects.count()` instead of `len(Facility.objects.all())`